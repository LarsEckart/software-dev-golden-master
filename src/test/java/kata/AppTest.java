package kata;

import java.util.List;
import java.util.Map;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
class AppTest {

  @Test
  void golden_master() {
    List<Performance> performances =
        List.of(
            new Performance("hamlet", 10),
            new Performance("as-like", 25),
            new Performance("othello", 20));

    Map<String, Play> plays =
        Map.of(
            "hamlet", new Play("Hamlet", "tragedy"),
            "as-like", new Play("As You Like It", "comedy"),
            "othello", new Play("Othello", "tragedy"));

    Invoice bigCo = new Invoice("BigCo", performances);
    String statement = new StatementPrinter().print(bigCo, plays);
    Approvals.verify(statement);
  }

  @Test
  void sampled_golden_master() {
    Integer[] tragedyAudience = {20, 29, 30, 31, 40};
    Integer[] comedyAUdience = {10, 19, 20, 21, 30};
    CombinationApprovals.verifyAllCombinations(
        this::print,
        tragedyAudience,
        comedyAUdience
    );
  }

  private String print(int tragedyAudience, int comedyAudience) {
    List<Performance> performances =
        List.of(
            new Performance("hamlet", tragedyAudience),
            new Performance("as-like", comedyAudience),
            new Performance("othello", 20));

    Map<String, Play> plays =
        Map.of(
            "hamlet", new Play("Hamlet", "tragedy"),
            "as-like", new Play("As You Like It", "comedy"),
            "othello", new Play("Othello", "tragedy"));

    Invoice bigCo = new Invoice("BigCo", performances);
    return new StatementPrinter().print(bigCo, plays);
  }
}
