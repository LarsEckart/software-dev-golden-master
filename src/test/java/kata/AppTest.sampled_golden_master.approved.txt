[20, 10] => Statement for BigCo
  Hamlet: $400.00 (20 seats)
  As You Like It: $330.00 (10 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,130.00
You earned 2 credits
 
[20, 19] => Statement for BigCo
  Hamlet: $400.00 (20 seats)
  As You Like It: $357.00 (19 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,157.00
You earned 3 credits
 
[20, 20] => Statement for BigCo
  Hamlet: $400.00 (20 seats)
  As You Like It: $360.00 (20 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,160.00
You earned 4 credits
 
[20, 21] => Statement for BigCo
  Hamlet: $400.00 (20 seats)
  As You Like It: $468.00 (21 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,268.00
You earned 4 credits
 
[20, 30] => Statement for BigCo
  Hamlet: $400.00 (20 seats)
  As You Like It: $540.00 (30 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,340.00
You earned 6 credits
 
[29, 10] => Statement for BigCo
  Hamlet: $400.00 (29 seats)
  As You Like It: $330.00 (10 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,130.00
You earned 2 credits
 
[29, 19] => Statement for BigCo
  Hamlet: $400.00 (29 seats)
  As You Like It: $357.00 (19 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,157.00
You earned 3 credits
 
[29, 20] => Statement for BigCo
  Hamlet: $400.00 (29 seats)
  As You Like It: $360.00 (20 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,160.00
You earned 4 credits
 
[29, 21] => Statement for BigCo
  Hamlet: $400.00 (29 seats)
  As You Like It: $468.00 (21 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,268.00
You earned 4 credits
 
[29, 30] => Statement for BigCo
  Hamlet: $400.00 (29 seats)
  As You Like It: $540.00 (30 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,340.00
You earned 6 credits
 
[30, 10] => Statement for BigCo
  Hamlet: $400.00 (30 seats)
  As You Like It: $330.00 (10 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,130.00
You earned 2 credits
 
[30, 19] => Statement for BigCo
  Hamlet: $400.00 (30 seats)
  As You Like It: $357.00 (19 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,157.00
You earned 3 credits
 
[30, 20] => Statement for BigCo
  Hamlet: $400.00 (30 seats)
  As You Like It: $360.00 (20 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,160.00
You earned 4 credits
 
[30, 21] => Statement for BigCo
  Hamlet: $400.00 (30 seats)
  As You Like It: $468.00 (21 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,268.00
You earned 4 credits
 
[30, 30] => Statement for BigCo
  Hamlet: $400.00 (30 seats)
  As You Like It: $540.00 (30 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,340.00
You earned 6 credits
 
[31, 10] => Statement for BigCo
  Hamlet: $410.00 (31 seats)
  As You Like It: $330.00 (10 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,140.00
You earned 3 credits
 
[31, 19] => Statement for BigCo
  Hamlet: $410.00 (31 seats)
  As You Like It: $357.00 (19 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,167.00
You earned 4 credits
 
[31, 20] => Statement for BigCo
  Hamlet: $410.00 (31 seats)
  As You Like It: $360.00 (20 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,170.00
You earned 5 credits
 
[31, 21] => Statement for BigCo
  Hamlet: $410.00 (31 seats)
  As You Like It: $468.00 (21 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,278.00
You earned 5 credits
 
[31, 30] => Statement for BigCo
  Hamlet: $410.00 (31 seats)
  As You Like It: $540.00 (30 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,350.00
You earned 7 credits
 
[40, 10] => Statement for BigCo
  Hamlet: $500.00 (40 seats)
  As You Like It: $330.00 (10 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,230.00
You earned 12 credits
 
[40, 19] => Statement for BigCo
  Hamlet: $500.00 (40 seats)
  As You Like It: $357.00 (19 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,257.00
You earned 13 credits
 
[40, 20] => Statement for BigCo
  Hamlet: $500.00 (40 seats)
  As You Like It: $360.00 (20 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,260.00
You earned 14 credits
 
[40, 21] => Statement for BigCo
  Hamlet: $500.00 (40 seats)
  As You Like It: $468.00 (21 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,368.00
You earned 14 credits
 
[40, 30] => Statement for BigCo
  Hamlet: $500.00 (40 seats)
  As You Like It: $540.00 (30 seats)
  Othello: $400.00 (20 seats)
Amount owed is $1,440.00
You earned 16 credits
 
